### PR TITLE
document GM API changes

### PR DESCRIPTION
**Pull request by tophf**
* add GM API changes in VM2.12.0
* add mini-TOC
* add and rephrase trivial details
* show the presence/absence of a returned value in js snippet
* tweak CSS: bold param name, less spacing between param name and description
* omit `parent` in `parent.child` param descriptions: GM functions only have one object at most so repeating its name is redundant, moreover the structure is obvious from the indent.

### CODE REVIEW

File: content/pages/api/gm.md [hunk 1]
```diff
@@ -80,277 +113,342 @@ GM_setValue(key, value)
...
-        - `Cookie`
-        - `Host`
-        - `Origin`
-        - `Referer`
-        - `User-Agent`
+        - `'Cookie'`
+        - `'Host'`
+        - `'Origin'`
+        - `'Referer'`
+        - `'User-Agent'`
```
Comment (tophf) [inline content/pages/api/gm.md]: Using quotes for consistency with the mini example above.

File: content/pages/api/gm.md [hunk 2]
```diff
@@ -80,277 +113,342 @@ GM_setValue(key, value)
...
 
-        One of `text`, `json`, `blob` and `arraybuffer`. The default value is `text`.
+        One of the following:
 
-    - `details.timeout` *number* *v2.9.5+*
+        - `'text'` *(default value)*
+        - `'json'`
+        - `'blob'`
+        - `'arraybuffer'`
+        - `'document'` *(since VM2.12.0)*
```
Comment (tophf) [inline content/pages/api/gm.md]: Using quotes to make the type of the param more obvious.

File: content/pages/api/gm.md [hunk 3]
```diff
@@ -6,69 +6,102 @@ path: "/api/gm/"
 
 `GM_*` are a group of special APIs provided by Violentmonkey.
 
+<div class="toc">
```
Comment (tophf) [inline content/pages/api/gm.md]: Maybe don't reduce the spacing (that's to reduce the "wasted" space), but instead just move the lists to the bottom of the page?

Comment (gera2ld) [reply content/pages/api/gm.md]: I'll try to find a way to generate TOC automatically. It's OK for now.

Comment (gera2ld) [reply content/pages/api/gm.md]: What do you mean by *the spacing*?

File: content/pages/api/index.md [hunk 4]
```diff
@@ -4,10 +4,9 @@ date: 2017-12-02 21:20:04
 path: "/api/"
 ---
 
-Violentmonkey APIs are derived from those in Greasemonkey v3, and most of them work the same way.
+Violentmonkey APIs are derived from those in Greasemonkey v3, and most of them work the same way.
```
Comment (gera2ld) [inline content/pages/api/index.md]: Redundant whitespace.

File: content/pages/api/gm.md (hunk 3 above)
Comment (tophf) [reply content/pages/api/gm.md]: I've reduced the inter-line spacing so the list is smaller.

Comment (gera2ld) [reply content/pages/api/gm.md]: Oh I see, that's OK.

**Pull request merged**